randovania/games/samus_returns/exporter/patch_data_factory.py

```diff
@@ -58,18 +115,78 @@ def _start_point_ref_for(self, node: Node) -> dict:
         else:
             return {}
 
+    def _key_error_for_node(self, node: Node, err: KeyError):
+        return KeyError(f"{self.game.region_list.node_name(node, with_region=True)} has no extra {err}")
+
+    def _level_name_for(self, node: Node) -> str:
+        region = self.game.region_list.nodes_to_region(node)
+        return region.extra["scenario_id"]
+
+    def _teleporter_ref_for(self, node: Node, actor_key: str = "actor_name") -> dict:
+        try:
+            return {
+                "scenario": self._level_name_for(node),
+                # "layer": node.extra.get("actor_layer", "default"),
+                "actor": node.extra[actor_key],
+            }
+        except KeyError as e:
+            return {}
+            raise self._key_error_for_node(node, e)
+
+    def _callback_ref_for(self, node: Node) -> dict:
```

Comment:
The whole method is unused afaik. We don't have such a thing at the current state of msr patcher.
Maybe we will need for metroids. Depends on what we will implement in the patcher.